ncompass_rust/trace_converters/nsys-chrome/tests/test_linker_adapters.rs

```
@@ -0,0 +1,181 @@
+
+    let result = adapter.get_time_range(&event);
+    assert!(result.is_none());
+}
+
+#[test]
+fn test_get_time_range_missing_end_ns() {
+    let adapter = NsysEventAdapter;
+    let event = ChromeTraceEvent::complete(
+        "kernel".to_string(),
+        100.0,
+        50.0,
+        "Device 0".to_string(),
+        "Stream 1".to_string(),
+        "kernel".to_string(),
+    )
+    .with_arg("start_ns", serde_json::json!(100000));
+
+    let result = adapter.get_time_range(&event);
+    assert!(result.is_none());
+}
+
+#[test]
+fn test_get_time_range_zero_duration() {
+    let adapter = NsysEventAdapter;
+    let event = ChromeTraceEvent::complete(
+        "kernel".to_string(),
+        100.0,
+        0.0,
+        "Device 0".to_string(),
+        "Stream 1".to_string(),
+        "kernel".to_string(),
+    )
+    .with_arg("start_ns", serde_json::json!(100000))
+    .with_arg("end_ns", serde_json::json!(100000));
+
+    let result = adapter.get_time_range(&event);
+    assert!(result.is_some());
+    let (start, end) = result.unwrap();
+    assert_eq!(start, 100000);
+    assert_eq!(end, 100000);
+}
+
+#[test]
+fn test_get_correlation_id_valid() {
+    let adapter = NsysEventAdapter;
+    let event = ChromeTraceEvent::complete(
+        "kernel".to_string(),
+        100.0,
+        50.0,
+        "Device 0".to_string(),
+        "Stream 1".to_string(),
+        "kernel".to_string(),
+    )
+    .with_arg("correlationId", serde_json::json!(12345));
+
+    let result = adapter.get_correlation_id(&event);
+    assert_eq!(result, Some(12345));
+}
+
+#[test]
+fn test_get_correlation_id_missing() {
+    let adapter = NsysEventAdapter;
+    let event = ChromeTraceEvent::complete(
+        "kernel".to_string(),
+        100.0,
+        50.0,
+        "Device 0".to_string(),
+        "Stream 1".to_string(),
+        "kernel".to_string(),
+    );
+
+    let result = adapter.get_correlation_id(&event);
+    assert!(result.is_none());
```

[QUESTION] just to check, we are on purpose kind of silently failing right? Or should these tests be raising errors?

---

intentionaal